arm-software/embedded/arm-runtimes/test-support/generic-test-wrapper.py

```diff
@@ -0,0 +1,153 @@
+        return EXIT_CODE_SKIP
+
+    # Some picolibc tests expect argv[0] to be literally "program-name", not
+    # the actual program name.
+    argv = ["program-name"] + args.arguments
+
+    parent_dir = os.path.dirname(args.image)
+    exeNameString = os.path.basename(args.image)
+
+    if args.qemu_command:
+        return run_qemu(
+            args.qemu_command,
+            args.qemu_machine,
+            args.qemu_cpu,
+            args.qemu_params.split(":") if args.qemu_params else [],
+            exeNameString,
+            argv,
+            None,
+            parent_dir,
+            args.verbose,
+        )
+    else:
+        return run_fvp(
+            args.fvp_install_dir,
+            args.fvp_config_dir,
+            args.fvp_model,
+            args.fvp_config,
+            exeNameString,
+            argv,
+            None,
+            parent_dir,
+            args.verbose,
+            args.tarmac,
+        )
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Run a single test using either qemu or an FVP"
+    )
+    # Top-level required argument
+    parser.add_argument("--test_executor", required=True, choices=["fvp", "qemu"],
+                        help="Executor type: fvp or qemu")
+
+    # Common arguments
+    parser.add_argument("--image", help="image file to execute")
+    parser.add_argument("--argv", help="parameters for executable")
+    parser.add_argument(
+        "--verbose",
+        action="store_true",
+        help="Print verbose output. This may affect test result, as the output "
+        "will be added to the output of the test.",
+    )
+
+    # FVP arguments
+    parser.add_argument(
+        "--fvp_install_dir", help="Directory in which FVP models are installed"
+    )
+    parser.add_argument(
+        "--fvp_config_dir", help="Directory in which FVP models are installed"
+    )
+    parser.add_argument(
+        "--fvp_model",
+        help="model name for FVP",
+    )
+    parser.add_argument(
+        "--fvp_config",
+        action="append",
+        help="FVP config file(s) to use",
+    )
+    parser.add_argument(
+        "--tarmac",
+        help="file to wrote tarmac trace to (FVP only)",
+    )
+
+    # QEMU arguments
+    parser.add_argument(
+        "--qemu_machine",
+        help="name of the machine to pass to QEMU",
+    )
+    parser.add_argument(
+        "--qemu_command",
+        help="name of the qemu binary used to execute. Must be present in PATH.",
+    )
+    parser.add_argument(
+        "--qemu_cpu", required=False, help="name of the cpu to pass to QEMU"
+    )
+    parser.add_argument(
+        "--qemu_params",
+        type=str,
+        help='list of arguments to pass to qemu, separated with ":"',
+    )
+
+    parser.add_argument(
+        "arguments",
+        nargs=argparse.REMAINDER,
+        default=[],
+        help="optional arguments for the image",
+    )
+    args = parser.parse_args()
+
+    # Conditional validation
```

Comment:
It is not clear what is the point or advantage of having this script with a bunch of executor specific options vs providing the same options to one of the underlying scripts: the caller logic will only differ in the name of the script - the rest will need to be worked out in the same way.